platform/src/main/java/org/hillview/storage/CassandraSSTableLoader.java

```diff
@@ -171,137 +230,269 @@ public Schema getSchema(CFMetaData metadata) {
+                                    col.appendMissing(); 
+                                } else {
+                                    switch ((CQL3Type.Native) colType) {
+                                        case ASCII:
+                                        case BOOLEAN:
+                                        case INET:
+                                        case TEXT:
+                                        case TIMEUUID:
+                                        case UUID:
+                                        case VARCHAR:
+                                            col.append(value.toString());
+                                            break;
+                                        case BLOB:
+                                            col.append(serializers.get(colType).toCQLLiteral(byteBuff).toString());
+                                            break;
+                                        case INT:
+                                            col.append((Integer) value);
+                                            break;
+                                        case SMALLINT:
+                                            col.append(((Short) value).intValue());
+                                            break;
+                                        case TINYINT:
+                                            col.append(((Byte) value).intValue());
+                                            break;
+                                        case VARINT:
+                                            col.append(((BigInteger) value).doubleValue());
+                                            break;
+                                        case BIGINT:
+                                            col.append(((Long) value).doubleValue());
+                                            break;
+                                        case DECIMAL:
+                                            if (bigDecimalInRange((BigDecimal) value))
+                                                col.append(((BigDecimal) value).doubleValue());
+                                            break;
+                                        case DOUBLE:
+                                            col.append((Double) value);
+                                            break;
+                                        case FLOAT:
+                                            col.append(((Float) value).doubleValue());
+                                            break;
+                                        case COUNTER:
+                                            col.append(CounterContext.instance().total(byteBuff));
+                                            break;
+                                        case TIME:
+                                            // TODO: Convert Time type to Java Time instead of Java Instant
+                                            long myTime = TimeSerializer.instance.deserialize(byteBuff);
```

Comment:
You already have the cassandra Time in the value. Why do you need to do this?

Reply:
To get the Long representation, because you were asking me to add 00-01-01 as the date. There is no method at Cassandra TimeType to get the Long representation directly, thus I deserialize the byteBuff. Please let me know if I misunderstood anything.